### ISSUE TITLE
inconsistent gradients with fd simulations

### ISSUE THREAD

**Issue by MarkMa1990** [NanoComp/meep#2106]
Hi all,

I have a question about the scaling factor for setting adjoint sources. I found it was set to meep.GaussianSource(fcen, fwidth=0.1*fcen) for the testing example (see code link below). then, I did a robust test by tuning this bandwidth factor. The scaling factor turns out to be very sensitive to the bandwidth. It seems that the adjoint field E_adj are really different by a phase factor. and the phase factor is not well considered for adjoint sources at several different bandwidth.

### Structure, monitor, and source setting
![image](https://user-images.githubusercontent.com/10673026/173834258-9af2127f-923a-4884-b9c2-8be9c0e7b02c.png)

### Test
df is a factor for the adjoint source profile meep.GaussianSource(fcen, fwidth=df*fcen).
![image](https://user-images.githubusercontent.com/10673026/173834390-bbf6a0f9-fe34-48e7-96e2-246105a058c5.png)
![image](https://user-images.githubusercontent.com/10673026/173835901-e04dc5ac-7d63-44b4-8c34-e744c69f13e3.png)

### Code in python library
![image](https://user-images.githubusercontent.com/10673026/173834664-ac7082bf-1ee4-4ea2-b4b0-027aa91b17a2.png)
![image](https://user-images.githubusercontent.com/10673026/173836104-4510503a-16a5-46a5-a570-9394d53884de.png)

### Meep version:  1.18.0, 1.19.0, 1.20.0, 1.21.0, 1.22.0, 1.23.0

**Comment by smartalecH** [comment]
Hmm as @mochen4 pointed out, I thought we fixed this in #1515...

I know our tests currently check at arbitrary bandwidths...

**Comment by smartalecH** [comment]
Sure but `test_adjoint_jax.py` still tests different values for the bandwidth.

**Comment by ianwilliamson** [comment]
Also see #1541.

Despite the tests checking different values for the forward source bandwidth, this still seemed to be an issue for me and I had trouble pinning it down.

**Comment by mochen4** [comment]
Actually, I realized that the issue is about the bandwidth of the adjoint source, i.e. the hardcoded 0.1 here:
https://github.com/NanoComp/meep/blob/619c879be3d53a20b882b4ca0443ea2158f03bd7/python/adjoint/objective.py#L102
In this case, #1515 and the tests are probably irrelevant, because they are about the bandwidth of the forward source.

I did some more tests, and found something similar to what was originally suggested in https://github.com/NanoComp/meep/issues/1507#issue-812520241. The additional phase for adjoint source is ``np.exp(2j * np.pi * cutoff / fwidth_frac)``, where ``cutoff`` is the another parameter for the Gaussian source and is default to 5. Also notice that this phase is the conjugate of the phase in https://github.com/NanoComp/meep/issues/1507#issue-812520241, which was about the bandwidth of the forward source.
